### ISSUE TITLE
EIR To Develop Community Middle Layer for Stacks (and beyond)

### ISSUE THREAD

**Issue by falkonprods** [stacksgov/residence-program#9]
**What's your name/your team's name?**
Name: Escape Labs, Team: Drew Falkman / David Collier

**Best email on which to contact you?**
drew@drewfalkman.com

**Personal Statement**
2021 has shown how community can be the difference between success and failure in crypto projects, especially NFTs and games.  While Ethereum has seen platforms emerge to support this like [Collab.land](Collab.land) and [temple.fans](temple.fans), the Stacks environment remains sorely lacking. Through this residency, we would like to test, build and validate features (using design thinking and lean methodologies) for an app that could solve these problems, launch an MVP platform to test and refine, and finish with a comprehensive middle layer for communities on Stacks. Key features include: the ability to authenticate users based on token ownership (NFT or fungible)  linked to Discord, the ability to send token drops, Discord engagement tools and  reporting features are also on the roadmap (see below). The end result will be to launch a “middle layer” app that can support developers getting their communities functioning quickly and integrated into their projects. Eventually this platform will cover the main aspects of community: membership, reputation, rewards and contributions. 

**End Product**
Over the course of this residency, we would contribute the following to the Stacks community:

- Roadmap and plans for a platform to address the primary needs in the Stacks developer and entrepreneur community in this opportunity area.
- A platform that would empower organizations to solve these problems.The platform would be comprised of three key elements: 
- An authorization methodology to integrate stacks wallet logins with Discord or other communities
- A hosted UI to manage communities and performing key activities - notably starting with coin drops to registered wallets/users. 
- Additional features as highlighted in the roadmap to increase engagement and support community managers' and projects' needs.

**Roadmap**
Overall, we are looking to analyze the following thesis: 

> Stacks needs a platform for creating communities that integrate on-chain elements (e.g. token holding) with off chain behaviors. Additionally, this need is barely understood and only partially addressed in decentralized projects on all blockchain projects leaving many untapped opportunities to help NFT, blockchain gaming and DeFi projects with tools to better understand, interact with and grow the communities that are central to their existence.

### Milestones:

![image](https://user-images.githubusercontent.com/1200120/139559707-051bd018-fa51-40a6-bd5a-64b422ef4643.png)

**Appendix**
### Long Term Platform Direction:

![image](https://user-images.githubusercontent.com/1200120/136590533-7ff86cf4-c756-40c8-949f-d7561e9d15b4.png)

Overall, community in crypto has some key pillars, we intend to address much of this as the platform evolves:

1. User management and membership. This is the ability to auth users to their wallet, and often verify they are holding some kind of token, fungible or NFT. This also includes profiles, a member directory, and possibly KYC or personhood verification of some kind.
2. Contributions (including consensus). As communities grow their will be a need to encourage contributions, and for certain types of communities having DAO-like features will be important: voting, spending, funding, and consensus-based administration.
3. Reputation. This isn’t just about having points or notes about someone in a profile, but is about understanding users in a community. Recognizing members through Discord roles, tracking users' participation in chat, social and other contributions (e.g. code, art, and other organizational support), and having users generally recognizable by admins and other members of the community for their contributions and areas of interest or expertise.
4. Rewards. These  can be in the form of token drops, social tokens, and more  - and can often be related to reputation and achievements, or simply being a member.

We plan on developing through solid partners who are building community currently, and have one high profile partner lined up already.

### A Community CRM and Community Management Console
One way to think of this platform is like a CRM for Discord/Telegram and it could eventually have other features for managing a real time community. We will discover more "killer app" features as we perform user research in this residency, work with partners to support their efforts, get feedback from the community, and track real world usage of the MVP (launching in 3 months). 

A sample of what this could look like (note this is just a sketch, not  the actual end product):

![image](https://user-images.githubusercontent.com/1200120/139559215-db9879eb-9979-4cf8-b210-9236a9ff2ca9.png)

Some future features could include:

- User profiles with metrics on engagement, posts, retention, ranking to help you identify and reward users
- Content marketing: setup a few posts in advance and these will be "drip fed" over time
- Gamified karma system - allow users to thank each other
- Quizzes/Polls/FAQs
- Event countdown
- Distribution of rewards based on rules
- Conversion funnel of users from interested to token owners to collaborators
- Invitation management on social media

**Some samples of Drew’s related work:**
DaoOS: https://github.com/syvita/daoOS
MiamiVoice (built on daoOS:): http://miamivoice.org 
	(won MiamiCoin Hackathon: https://www.citycoins.co/post/announcing-the-winners-of-miamicoin-makers-month) 

Some samples of DC’s related work: 
- https://dc.rik.ai/ 
- https://github.com/dcsan

**Comment by owenstrevor** [comment]
This kind of a product is not only needed by every NFT and DeFi project in order to manage and grow their community, it's sorely needed. Crypto is centered around community building, and a project is only as good as its community. Almost every NFT project I've seen on Ethereum uses Collab.land to handle user authentication in discord, so I'm confident this research is not only addressing a big pain point, but also a large market that can lift all boats in our ecosystem.

**Comment by Bowtiedcheetah** [comment]
Love this idea. Phases of Satoshi here, we have been desperately looking for something like this. Were a small but smart community and would love to help and get involved

**Comment by falkonprods** [comment]
WGMI!

**Comment by jennymith** [comment]
For all community members interested in keeping up with Escape Labs's residency, check out their upcoming plans and follow this thread for weekly updates from @falkonprods and @dcscan themselves:

### Scope of Work

**Background**

2021 has shown how community can be the difference between success and failure in crypto projects, especially NFTs and games. While Ethereum has seen platforms emerge to support this like Collab.land and temple.fans, these tools still only solve part of the issue, and the Stacks environment remains lacking in these types of tools. Through this residency, we would like to test, build and validate features (using design thinking and lean methodologies) for an app that could solve these problems, launch an MVP platform to test and refine, and finish with a comprehensive middle layer for communities on Stacks. Key features include: the ability to authenticate users based on token ownership (NFT or fungible) linked to Discord, the ability to send token drops, Discord engagement tools and reporting features are also on the roadmap (see below). The end result will be to launch a “middle layer” app that can support developers getting their communities functioning quickly and integrated into their projects. Eventually this platform will cover the main aspects of community: membership, reputation, rewards and contributions.

**Project Thesis**

Stacks needs a platform for creating communities that integrate on-chain elements (e.g. token holding) with off chain behaviors. Additionally, this need is barely understood and only partially addressed in decentralized projects on all blockchain projects leaving many untapped opportunities to help NFT, blockchain gaming and DeFi projects with tools to better understand, interact with and grow the communities that are central to their existence.

**Roadmap for first 3 months:**

In the first three months of our residency, we plan on engaging projects on Stacks that are currently managing communities. We will conduct surveys and perform user research to better understand the problems they are facing and their current solutions. We will use best practices design thinking and product management techniques to design and spec out a community app that will do the following:

1. A connector that will enable users to link their STX address to their Discord ID via Hiro Web Wallet. This will be in the vein of what Collab.land does on ETH. Typing a command into Discord will have our bot send them a DM with a link unique to them. On clicking, they will be prompted to connect with their Hiro Wallet. Connecting will let them choose a wallet address, then the user will be completed and the project will have a record of the user’s Discord ID and their wallet.
2. Once we have users’ wallet addresses associated with their Discord ID, we can assign roles to them (e.g. “whitelist confirmed”), and even assign channels to them based on their holdings - holding an NFT in a collection for example, or a social token will grant them access to the #CLUB channel, etc.
3. Along the same lines, a community administrator can either export the wallet addresses to add to a smart contract (like a whitelist contract to enable users with those addresses to mint an NFT), or perform airdrops of tokens or NFTs to wallets in their collection.
4. Finally, to support engagement of communities within Discord, we are investigating creating interactive bots and have some ideas in the form of games and interactive chatbots. We plan on experimenting and launching with one of these tools.

If you're interested in working with Escape Labs to activate your Stacks community, DM them on Twitter at @esc_labs.

**Comment by falkonprods** [comment]
Weekly update 2/4:

Note: Drew and DC have decided to go in different directions.

**From Drew:** We have launched our web site at [frens.place](https://frens.place)! This comes with a full re-branding to _frens_. This re-branding also comes along with some fixes and updates to both simplify and enhance what we had previously with Layer C.

- We currently have the ability to verify your STX address, and be assigned a discord role when verified.
- We also have the ability to assign a discord role to whomever is holding an NFT that matches the contract of each guild.
- The export of wallets can be used for whitelisting or drops.

In the next weeks, the following items are on our roadmap:

- Integration with CAPTCHA bot and invite tracking bot (that auto-assigns roles at certain numbers of invites) to alleviate the need to install a bunch of plugins and dealing with conflicts, as well as manual work assigning roles
- A web UI for admins, and for members to control this tool 
- An analytics dashboard for admins
- Discord bots to enhance engagement of discord communities.
- Portable reputations for members that highlight a user's skills and community involvement
- Easier drops and rewards for admins through the web UI

Our near-term product vision for _frens_ is to empower communities to GROW by providing tools that save mods time so they can spend more time interacting with the community, as well as providing better tools that encourage engagement in the community Discord.

**Comment by dcsan** [comment]
From @dcsan the project has been forked. 
The first version of the project was built and launched at https://stan.escapelabs.xyz/
Also launched the first generative art project on stacks https://www.dcards.art/

My focus is now on the next generation of interactive and social NFTs. I've launched AlfaLabs to bring this to reality
https://www.alfalabs.xyz

# Bots Roadmap
The roadmap for the Bot and community side:
https://www.alfalabs.xyz/bots
We built the[LayerC](https://stan.escapelabs.xyz/)bot to allow Stacks projects to verify their users and manage whitelists, mainly pre-launch activities to support the mint-day. As Stacks projects mature, we're now shifting to support later points in the user journey. We're building the next generation community bot AlfaBot to enable more interactivity and gamification, deeper integration with the Stacks blockchain and a plugin system for new features.
<img width="737" alt="image" src="https://user-images.githubusercontent.com/514002/152918910-f5ed96cf-4a88-4d7c-9a66-0f54920a2d9f.png">

# AlfaBlox
Additionally we are planning out AlfaBlox which are components to help accelerate stacks project development.
https://www.alfalabs.xyz/blox

The first two blox are in development and we plan to open source them.

Very interested in working with other Stacks NFT projects who are building interactive Stacks NFTs, dApps and game projects. 

Feel free to mail me dc@alfalabs.xyz or join our Discord!
https://discord.gg/3XpsMPXuN2

❤️  from `[α]LABS`

**Comment by falkonprods** [comment]
Weekly update 2/10:

**From Drew:** 

The _frens_ user base is growing as is our team. Four new customers. We have also added a new dev, which brings us to two full-time developers in addition to a couple contractors. We look forward to ramping up and having quicker turnaround on new features, upgrades and bug fixes.

We have prepped our backlog and designs for the first full team sprint, and it includes UI improvements, support for multiple NFT contracts and multiple role levels to assign based on NFT holdings. Additionally, we are going to be performing some re-factoring of the code to make it easier for a larger team to build quickly, and are working on designs (which we will share soon) for the next iteration.

Drew was featured this week on Built on Bitcoin podcast, episode 41 (https://www.builtonbitcoin.xyz/episodes/), and will be speaking more and getting the word out about our app and supporting communities developing on Stacks. Additionally, we have agreed on a partnership with one of the larger projects in Stacks...more on this to come.

We have a goal of making the _frens_ Discord THE place to come and discuss and learn about Web3 community development and growth. Please come and join us: https://discord.gg/6EMExFbS

**Comment by falkonprods** [comment]
Update 2/14:

**From Drew:**

We have hired a new developer. We are re-architecting the code to be able to onboard more quickly, have a more scalable server environment, add CI/CD pipeline, add more unit tests and more. We have made a few minor improvements and are starting to get into our backlog.

We have our backlog ready for the next couple months of development and are moving from wireframes to designs for the UI which we have laid the groundwork for in a new repo.

Drew was featured on 0Authority streaming, and gave a talk to the cohort on community development and basic Discord setup and configuration.

Drew created a template for Web3 communities to brainstorm and align on their community development available for anyone to use: https://www.figma.com/file/04mcKKDu0QUE9lt2lLPq83/Web3-Community-Model-Canvas?node-id=0%3A1
